---
title: "コンパイラ エラー CS1017"
ms.date: 07/20/2015
ms.prod: .net
ms.technology: devlang-csharp
ms.topic: article
f1_keywords: CS1017
helpviewer_keywords: CS1017
ms.assetid: e0902e8a-b042-4711-a8a6-83456a3f88cb
caps.latest.revision: "9"
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 5170c43a4618348c2dea1a0dade2f6d6ddbeb2d2
ms.sourcegitcommit: bd1ef61f4bb794b25383d3d72e71041a5ced172e
ms.translationtype: MT
ms.contentlocale: ja-JP
ms.lasthandoff: 10/18/2017
---
# <a name="compiler-error-cs1017"></a><span data-ttu-id="827da-102">コンパイラ エラー CS1017</span><span class="sxs-lookup"><span data-stu-id="827da-102">Compiler Error CS1017</span></span>
<span data-ttu-id="827da-103">catch 句を、try ステートメントの一般的な catch 句の後に置くことはできません。</span><span class="sxs-lookup"><span data-stu-id="827da-103">Catch clauses cannot follow the general catch clause of a try statement</span></span>  
  
 <span data-ttu-id="827da-104">パラメーターを一切受け取らない `catch` ブロックは、一連の `catch` ブロックの最後に配置する必要があります。</span><span class="sxs-lookup"><span data-stu-id="827da-104">A `catch` block that does not take any parameters must be the last in a series of `catch` blocks.</span></span> <span data-ttu-id="827da-105">例外の詳細については、「 [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md)」を参照してください。</span><span class="sxs-lookup"><span data-stu-id="827da-105">For more information on exceptions, see [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="827da-106">例</span><span class="sxs-lookup"><span data-stu-id="827da-106">Example</span></span>  
 <span data-ttu-id="827da-107">次の例では CS1017 が生成されます。</span><span class="sxs-lookup"><span data-stu-id="827da-107">The following sample generates CS1017:</span></span>  
  
```  
// CS1017.cs  
using System;  
  
namespace x  
{  
    public class b : Exception  
    {  
    }  
  
    public class a  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
  
            catch   // CS1017, must be last catch  
            {  
            }  
  
            catch(b)  
            {  
                throw;  
            }  
        }  
    }  
}  
```
